### ISSUE TITLE
Synchronize actor start up in lib.rs

### ISSUE THREAD

**Issue by eadventurous** [hyperledger/iroha#1363]
Currently main Iroha actors start asynchronously - meaning when the method `start` is called it spawns a new task. Therefore several actor's start up code runs in parallel. This leads to the following problems:
1. Actors can sometimes subscribe to messages later then they are sent. (Example Kura Init message to Sumeragi)
2. Some things like Torii should only be started when Sumeragi is listening to messages and is fully initialized. So that health endpoint is only available when Sumeragi is ready to receive messages.

**Comment by eadventurous** [comment]
First noticed in #1348
